File: ungs-docsys-ms/src/main/java/com/ungs/docsys/dtos/LanguageRequestDto.java
package com.ungs.docsys.dtos;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class LanguageRequestDto {
    @NotBlank(message = "Language must not be blank")
    @Size(max = 50, message = "Language must not exceed 50 characters")
    private String language;
    @NotBlank(message = "Level must not be blank")
    @Size(max = 50, message = "Level must not exceed 50 characters")
    private String level;
}

File: ungs-docsys-bff/src/modules/resume-user/dtos/language-request.dto.ts
import {
  IsOptional,
  IsString,
  MaxLength
} from 'class-validator';

export class LanguageRequestDto {
  @IsOptional()
  @IsString()
  @MaxLength(50, { message: 'Language must not exceed 50 characters' })
  language?: string;

  @IsOptional()
  @IsString()
  @MaxLength(50, { message: 'Level must not exceed 50 characters' })
  level?: string;
}